### PR TITLE
#2 Navbar

### PR DESCRIPTION
**Pull request by Cleop**
#2 Creates navbar with colour changing js for both desktop and mobile, ensures passing pa11y and adds Cypress tests
#47 Fixes tachyons media query bug

### CODE REVIEW

File: public/stylesheets/style.min.css [hunk 1]
```diff
@@ -1 +1,7405 @@
...-m{margin-right:var(--spacing-medium)}.mr4-m{margin-right:var(--spacing-large)}.mr5-m{margin-right:var(--spacing-extra-large)}.mr6-m{margin-right:var(--spacing-extra-extra-large)}.mr7-m{margin-right:var(--spacing-extra-extra-extra-large)}.mb0-m{margin-bottom:var(--spacing-none)}.mb1-m{margin-bottom:var(--spacing-extra-small)}.mb2-m{margin-bottom:var(--spacing-small)}.mb3-m{margin-bottom:var(--spacing-medium)}.mb4-m{margin-bottom:var(--spacing-large)}.mb5-m{margin-bottom:var(--spacing-extra-large)}.mb6-m{margin-bottom:var(--spacing-extra-extra-large)}.mb7-m{margin-bottom:var(--spacing-extra-extra-extra-large)}.mt0-m{margin-top:var(--spacing-none)}.mt1-m{margin-top:var(--spacing-extra-small)}.mt2-m{margin-top:var(--spacing-small)}.mt3-m{margin-top:var(--spacing-medium)}.mt4-m{margin-top:var(--spacing-large)}.mt5-m{margin-top:var(--spacing-extra-large)}.mt6-m{margin-top:var(--spacing-extra-extra-large)}.mt7-m{margin-top:var(--spacing-extra-extra-extra-large)}.mv0-m{margin-top:var(--spacing-none);margin-bottom:var(--spacing-none)}.mv1-m{margin-top:var(--spacing-extra-small);margin-bottom:var(--spacing-extra-small)}.mv2-m{margin-top:var(--spacing-small);margin-bottom:var(--spacing-small
```
[the remark this answers is not in the document]
Comment (Cleop) [reply public/stylesheets/style.min.css]: Good idea, I'll put it .gitignore 👍

[the remark this answers is not in the document]
Comment (nelsonic) [reply public/stylesheets/style.min.css]: @Cleop thanks.

[the remark this answers is not in the document]
Comment (SimonLab) [reply public/stylesheets/style.min.css]: Unfortunately I think we need to keep this css file.
The file is indeed generated with the npm command `css-compile` on pre-commit:
https://github.com/charterindex/application/blob/79aebccfb31560c804bf224c49224a1be15e6e2b/package.json#L51
This make sure that each time we push to a branch the last version of the css is compiled.
If the css file is ignored, we could possibly generate it on Heroku when we deploy but if I remember correctly files created on Heroku are saved only for a short period of time.

**Comment by nelsonic** [comment]
Thanks @Cleop 🥇

File: public/stylesheets/style.min.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (nelsonic) [reply public/stylesheets/style.min.css]: @SimonLab I think we can have a build-step on Heroku that will generate the static asset.
Also, recall that the "Production" version of the app will be on DigitalOcean where static files will not be "_ephemeral_" like on Heroku. 😉

**Comment by nelsonic** [comment]
@SimonLab please see: https://github.com/charterindex/application/issues/52#issuecomment-377241305 ❤️

**Comment by SimonLab** [comment]
@nelsonic I've added the css file to gitignore and also added a Procfile to compile the css before running the server. The PR should be ready now.

**Comment by nelsonic** [comment]
@SimonLab _almost_, adding the file that is already "staged" to `.gitignore` does not _remove_ it from the PR. (_i.e. it will continue to be tracked unless you `git rm` it_...)

**Comment by nelsonic** [comment]
@Cleop / @SimonLab we may need to do the compilation on CI prior to running the build...?

**Pull request merged**